backend/layers/business/business.py

```diff
@@ -147,6 +147,39 @@ def get_published_collection_version(self, collection_id: CollectionId) -> Optio
         """
         return self.database_provider.get_collection_mapped_version(collection_id)
 
+    def get_latest_published_collection_versions_by_schema(
+        self, schema_version: str
+    ) -> List[CollectionVersionWithPublishedDatasets]:
+        """
+        Returns a list with the latest published collection version that matches the given schema_version, for each
+        canonical collection
+        """
+        has_wildcards = "_" in schema_version
+        collection_versions = self.database_provider.get_collection_versions_by_schema(schema_version, has_wildcards)
+
+        # for each published canonical collection, map its most recently published collection version
+        collections = dict()
+        for collection_version in collection_versions:
+            if collection_version.published_at is None:
+                continue
+            canonical_collection_id = collection_version.collection_id.id
+            if canonical_collection_id not in collections:
+                collections[canonical_collection_id] = collection_version
+            else:
+                current_collection_version = collections[canonical_collection_id]
```

Comment:
`current` is a confusing prefix since your are iterating over the collections. It makes me think it's the current collection_version in this iteration. Maybe use `last`, or don't rename it and just use `collections[canonical_collection_id]` bellow.